translator/util/ec2util/ec2util.go

```diff
@@ -67,46 +71,51 @@ func initEC2UtilSingleton() (newInstance *ec2Util) {
 		log.Println("W! [EC2] Sleep until network is up")
 		time.Sleep(1 * time.Second)
 	}
+
 	if !networkUp {
 		log.Println("E! [EC2] No available network interface")
 	}
 
-	ses, err := session.NewSession()
+	err := newInstance.deriveEC2MetadataFromIMDS()
+
 	if err != nil {
-		log.Println("E! [EC2] getting new session info: ", err)
-		return
+		log.Println("E! [EC2] Cannot get EC2 Metadata from IMDS:", err)
 	}
-	md := ec2metadata.New(ses)
 
-	if !md.Available() {
-		log.Println("E! ec2metadata is not available")
-		return
-	}
+	return
+}
 
-	if info, err := md.GetMetadata("instance-id"); err == nil {
-		newInstance.InstanceID = info
-	} else {
-		log.Println("E! getting instance-id from EC2 metadata fail: ", err)
+func (e *ec2Util) deriveEC2MetadataFromIMDS() error {
+	ses, err := session.NewSession()
+
+	if err != nil {
+		return err
 	}
 
-	if info, err := md.GetMetadata("hostname"); err == nil {
-		newInstance.Hostname = info
-	} else {
-		log.Println("E! getting hostname from EC2 metadata fail: ", err)
+	md := ec2metadata.New(ses)
+
+	if !md.Available() {
+		return errors.New("EC2 metadata is not available.")
 	}
 
-	if info, err := md.GetMetadata("local-ipv4"); err == nil {
-		newInstance.PrivateIP = info
+	// Only need the API to scrape HostName
+	// More information on API: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html#instance-metadata-ex-2
+	if hostname, err := md.GetMetadata("hostname"); err == nil {
+		e.Hostname = hostname
 	} else {
-		log.Println("E! getting local-ipv4 from EC2 metadata fail: ", err)
+		log.Println("E! [EC2] Fetch hostname from EC2 metadata fail:", err)
 	}
 
-	if info, err := md.GetInstanceIdentityDocument(); err == nil {
-		newInstance.Region = info.Region
-		newInstance.AccountID = info.AccountID
+	// Only need the API to scrape Region, AccountId, PrivateIp, Instance ID
```

Comment:
Similarly, I don't see a lot of value from this comment. I understand that there are other attributes that are returned in the identity document, but it is clear what is extracted from the response, directly from the code below. Parroting that information in this comment means that if we add/remove attributes that we store, we also have to update this comment, even if the code already provides enough clear context.

Reply:
Agree with the same though on previous one.